### DISCUSSION TITLE
Confused about :npm, :onlyNpm, enabledManagers

### DISCUSSION THREAD

**Discussion by gremo** [renovatebot/renovate#9527] [Request Help] [answered]
After reading the documentation I'm a bit confused about `:npm` and `:onlyNpm`. I'll copy/paste the two presets:

`:npm`
_Keep package.json npm dependencies updated_

```json
{
  "npm": {
    "enabled": true
  }
}
```

`:onlyNpm`
_Renovate only npm dependencies_

```json
{
  "docker": {
    "enabled": false
  },
  "meteor": {
    "enabled": false
  }
}
```

**(1)** If I understand correctly, all managers are enabled by default. So what's the effect of setting `:npm` and why this preset exists in the first place? **(2)** Why `:onlyNpm` doesn't list other managers as well (for example, composer)? **(3)** Isn't that present the same as setting the following?

```json
{
  "enabledManagers": ["npm"]
}
```

Really thanks for your time.

**Comment by rarkins** [comment, marked as answer]
Those presets are outdated / incorrect now. They were originally from early days when only npm, meteor and Docker were supported and need updating.

**Reply by gremo** [reply]
So documentation should be updated? What you suggest for enable only `package.json` and NPM  manager?

**Reply by rarkins** [reply]
I think better to fix it or deprecate it than simply update the documentation. I recommend you use enabledManagers instead
